core/src/test/java/org/helsinki/paytrail/request/payments/PaytrailPaymentCreateTest.java

```diff
@@ -5,73 +5,102 @@
+        PaytrailClient client = new PaytrailClient(merchantId, secretKey);
 
-        TreeMap<String, String> headers = new TreeMap<>();
+        PaytrailPaymentCreateRequest.CreatePaymentPayload payload = new PaytrailPaymentCreateRequest.CreatePaymentPayload();
 
-        headers.put("checkout-account", merchantId);
-        headers.put("checkout-algorithm", "sha256");
-        headers.put("checkout-method", "POST");
-        headers.put("checkout-nonce", "564635208570151");
-        headers.put("checkout-timestamp", "2018-07-06T10:01:31.904Z");
-        //headers.put("content-type", "application/json; charset=utf-8");
+        payload.setStamp("d2568f2a-e4c6-40ba-a7cd-d573382ce549");
 
-        PaytrailClient client = new PaytrailClient(merchantId, secretKey);
+        payload.setReference("3759170");
+        payload.setAmount(1525);
+        payload.setCurrency("EUR");
+        payload.setLanguage("FI");
+        PaymentItem paymentItem = new PaymentItem();
+
+        paymentItem.setUnitPrice(1525);
+        paymentItem.setUnits(1);
+        paymentItem.setVatPercentage(24);
+        paymentItem.setProductCode("#1234");
+        ArrayList<PaymentItem> items1 = new ArrayList<>();
+        items1.add(paymentItem);
+        payload.setItems(items1);
+
+        PaymentCustomer customer = new PaymentCustomer();
+        customer.setEmail("severi.kupari@ambientia.fi");
+        payload.setCustomer(customer);
+
+        PaymentCallbackUrls callbackUrls = new PaymentCallbackUrls();
+
+        callbackUrls.setSuccess("https://ecom.example.com/cart/success");
+        callbackUrls.setCancel("https://ecom.example.com/cart/cancel");
 
-        Gson gson = new Gson();
+        payload.setCallbackUrls(callbackUrls);
+        payload.setRedirectUrls(callbackUrls);
 
-        LinkedHashMap<String, Object> object = new LinkedHashMap<>();
-        ArrayList<Object> items = new ArrayList<>();
-        LinkedHashMap<String, Object> item = new LinkedHashMap<>();
-        LinkedHashMap<String, Object> customerEmail = new LinkedHashMap<>();
-        LinkedHashMap<String, Object> redirectUrls = new LinkedHashMap<>();
+        PaytrailPaymentCreateRequest request = new PaytrailPaymentCreateRequest(payload);
+        CompletableFuture<PaytrailPaymentCreateResponse> response = client.sendRequest(request);
 
-        PaytrailPaymentCreateRequest.CreatePaymentPayload payload = new PaytrailPaymentCreateRequest.CreatePaymentPayload();
+        PaytrailPaymentCreateResponse paymentCreateResponse = mapper.to(response.get());
 
-        payload.setStamp("d2568f2a-e4c6-40ba-a7cd-d573382ce548");
+        try {
+            Gson gson = new Gson();
+            String json = gson.toJson(response.get());
+            System.out.println(json);
+            System.out.println(paymentCreateResponse.toString());
+        } catch (InterruptedException | ExecutionException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
Could you add some assetions to returning json? Like the values you give are returned in response?

Reply:
Added some basic assertions, but returning data does not contain any of the provided request params, so there is no way to reflect returning data on request data.